search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java

```diff
@@ -288,4 +314,20 @@ private Long distinctCountUsageMetricsFor(final String event, final UsageMetrics
             .findAll(filter, io.camunda.webapps.schema.entities.operate.UsageMetricsEntity.class);
     return metrics.stream().map(UsageMetricsEntity::value).distinct().count();
   }
+
+  private UserQuery injectUsernamesIntoQuery(
+      final Set<String> usernames, final UserQuery userQuery) {
+    return UserQuery.of(
```

Comment:
This can be a bit more robust with a `toBuilder()...build()` pattern. I've tried that for roles in https://github.com/camunda/camunda/pull/25555/files#diff-c7ee74abafdf2e69d0afefb6b2c26ec904c4b8b7534592b6070f550c8524ffeeR312.

Reply:
Good suggestion, changed now